src/lex.cpp

```diff
@@ -628,18 +632,22 @@ void lexer::parse_binary_number() {
     input += 1;
   }
 
+  report_garbage_in_number_literal(input);
+}
+
+void lexer::report_garbage_in_number_literal(const char8* input) {
   const char8* garbage_begin = input;
   for (;;) {
     switch (*input) {
-    QLJS_CASE_DECIMAL_DIGIT:
-    QLJS_CASE_IDENTIFIER_START:
-      input += 1;
-      break;
+      QLJS_CASE_DECIMAL_DIGIT:
+      QLJS_CASE_IDENTIFIER_START:
+        input += 1;
+        break;
       default:
         goto done_parsing_garbage;
     }
   }
-done_parsing_garbage:
+  done_parsing_garbage:
   const char8* garbage_end = input;
   if (garbage_end != garbage_begin) {
     this->error_reporter_->report(error_unexpected_characters_in_number{
```

Comment:
Line 658: report_garbage_in_number_literal accepts the input pointer, the assigns to `this->input_` at the end. This is different from every other function in this file; functions either exclusively manipulate `this->input_`, or they accept the input pointer and return the modified input pointer. I think report_garbage_in_number_literal should do one or the other. For example:

```cpp
const char8* lexer::report_garbage_in_number_literal(const char8* input) {
```

Reply:
Updated in latest commit.